apps/meteor/app/channel-settings/server/methods/saveRoomSettings.js

```diff
@@ -294,7 +296,8 @@ Meteor.methods({
 			settings = { encrypted: settings.encrypted };
 		}
 
-		if (room.broadcast && (settings.readOnly || settings.reactWhenReadOnly)) {
+		const checkReadOnly = Object.keys(settings).includes('readOnly') || Object.keys(settings).includes('reactWhenReadOnly');
```

Comment:
just ignore if you think it's not really important in this case, but could you assign `Object.keys(settings)` to one single variable? the way it's done we're reading the fields twice.

```suggestion
                const settingsKey = Object.keys(settings);
		const checkReadOnly = settingsKey.includes('readOnly') || settingsKey.includes('reactWhenReadOnly');
```